### ISSUE TITLE
"right-align" bug in editor

### ISSUE THREAD

**Issue by Mehran7kl** [markusfisch/ShaderEditor#111]
Whenever i begin a newline this happens
![Screenshot_۲۰۲۲۰۲۲۵_۱۷۰۹۰۲](https://user-images.githubusercontent.com/88193952/155724987-89e24865-3058-4286-97cc-4b9bb4cd829f.jpg)
Note: default language in my device is Persian and it's right-aligned.
it may help you to fix the bug.
And please add some methods to give us control over vertex shaders.
Thanks for creating this beautiful app.

**Comment by Mehran7kl** [comment]
I have changed language to English and now it's working fine but please fix the main issue.

**Comment by markusfisch** [comment]
Sorry for the inconvenience 😬 And yes, your hunch is right: the reason for this bug is that the editor takes its alignment from the system 🙈 Will fix it.

Thanks for reporting! 👍

**Comment by markusfisch** [comment]
Fixed in https://github.com/markusfisch/ShaderEditor/commit/0801067307a9437dee924dfd0516ed8563675aa3

**Comment by Mehran7kl** [comment]
Thanks.
Please release new version in Play Store.

**Comment by markusfisch** [comment]
Version 2.23.3 is in beta and will soon be available.

If you want to get it now, you can also get it directly from GitHub:https://github.com/markusfisch/ShaderEditor/releases/tag/2.23.3

**Comment by Mehran7kl** [comment]
the bug now is strange behaviour of space and tab and curly braces are bad aligned
see the image below
![Screenshot_20220302_000737_de markusfisch android shadereditor](https://user-images.githubusercontent.com/88193952/156245305-9f246edf-86e0-4fc0-ae53-15664020ccd0.jpg)
the problem with space and tab is :
they don't change the place of character selector after beginning a newline.
i can't explain more exact.

**Comment by markusfisch** [comment]
Hmm 🤔 I need to do some tests to find out what's happning here.

**Comment by Mehran7kl** [comment]
You can explicitly set APP Language/Locale to English. This issue has been fixed in some other apps by this method.

**Comment by Mehran7kl** [comment]
Haven't you fixed the problem yet?!
The problem is really annoying.
Please simply add an option to change the locale or explicitly set it.
the problem is that white spaces are still right-aligned. 
Best wishes.

**Comment by Mehran7kl** [comment]
Finally i fixed the bug myself 😁. Because you was locked the apk from Play Store and F-Droid. I used APK file that you have uploaded before in this page. i edited a line in manifest file containing "supportsRtl:true" to false. i first tested the same apk without changing and it had the bug; after changing the bug fixed. so we know that at least changing this attribute to false can fix the bug. so don't forget this in next update please. best wishes.

**Comment by Mehran7kl** [comment]
Finally i fixed the bug myself 😁. Because you was locked the apk from Play Store and F-Droid. I used APK file that you have uploaded before in this page. i edited a line in manifest file containing "supportsRtl:true" to false. i first tested the same apk without changing and it had the bug; after changing the bug fixed(seams OS doesn't force the app to use rtl anymore. actually in rtl systems(e.g my mobile with persian) shows everything from right to left. now i see the app options as you see😂) . so we know that at least changing this attribute to false can fix the bug. so don't forget this in next update please. best wishes.

**Comment by markusfisch** [comment]
Ah, okay! Good catch and good to know! 👍 

I'm still a little confused about `supportsRtl`. The docs say this flag declares whether an app is willing to support right-to-left (RTL) layouts, and I enabled it because I assumed, this would make the app fit in on RTL setups. But I have no experience with RTL layouts, so I really appreciate your feedback on this!

Thanks a lot for pointing this out and fixing the bug! I already changed it (see https://github.com/markusfisch/ShaderEditor/commit/d4aa8c246e207199a8b4fedbf6ee5f352a842290) and there will be an update soon.
